mantidproject/mantid
# Remove First Last From ALC Interface And Replace With Single Runs Box

**Description of work.**
Updated the ALC interface to now only have a single input box for runs instead of a first and a last one. In doing, the auto checkbox has been changed to now only find the last run if possible and no longer watches for files to be added and removed from the data directory. This has been confirmed to be an okay change from Stephen Cottrell, if the feature is requested again it can be added back in as a separate issue.

**NOTE:** This work was rebased off this other pull request #29185 which has been reviewed separately so ignore changes made from before **Aug 26**

**To test:**

1. Set default instrument to **MUSR**
2. Open the ALC Interface (Interfaces->Muon->ALC)
3. Enter into runs `15189-15191` and hit enter. Log combo box should be updated and you can now press load
4. Try checking auto. With the same input as before auto should change runs to `15189-15193,15195`. It is changed to this as it has found 15195 to be the most recent run in the directory of 15189, and run number 15194 does not exist in this directory so it has been excluded. 
5. Try pressing load again now
6. Hopefully no errors have come up so far, feel free to try different variations as input into runs. There are mocking tests which try to test it fully.

Fixes #24839  

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
